crates/bevy_render/src/renderer/render_device.rs

```diff
@@ -64,29 +73,48 @@ impl RenderDevice {
                         })
                 }
             }
-            // SAFETY: we are interfacing with shader code, which may contain undefined behavior,
-            // such as indexing out of bounds.
-            // The checks required are prohibitively expensive and a poor default for game engines.
-            // TODO: split this method into safe and unsafe variants, and propagate the safety requirements from
-            // https://docs.rs/wgpu/latest/wgpu/struct.Device.html#method.create_shader_module_trusted to the unsafe form.
+            // SAFETY:
+            //
+            // creates a shader module with user-customizable runtime checks which allows shaders to
+            // perform operations which can lead to undefined behavior like indexing out of bounds,
+            // thus it's the caller responsibility to pass a shader which doesn't perform any of this
+            // operations.
```

Comment:
i'd copy the short safety comment below, not much use in copying the long version from the top of the function.